drivers/sensor/paw3212/paw3212.c

```diff
@@ -0,0 +1,524 @@
+			LOG_INF("PAW3212 initialized");
+		} else {
+			k_delayed_work_submit(&dev_data->init_work,
+					      async_init_delay[dev_data->async_init_step]);
+		}
+	}
+}
+
+static int paw3212_init_cs(struct paw3212_data *dev_data)
+{
+	int err;
+
+	dev_data->cs_gpio_dev =
+		device_get_binding(CONFIG_PAW3212_CS_GPIO_DEV_NAME);
+	if (!dev_data->cs_gpio_dev) {
+		LOG_ERR("Cannot get CS GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->cs_gpio_dev,
+				 CONFIG_PAW3212_CS_GPIO_PIN,
+				 GPIO_DIR_OUT);
+	if (!err) {
+		err = spi_cs_ctrl(dev_data, false);
+	} else {
+		LOG_ERR("Cannot configure CS PIN");
+	}
+
+	return err;
+}
+
+static int paw3212_init_irq(struct paw3212_data *dev_data)
+{
+	int err;
+
+	dev_data->irq_gpio_dev =
+		device_get_binding(CONFIG_PAW3212_IRQ_GPIO_DEV_NAME);
+	if (!dev_data->irq_gpio_dev) {
+		LOG_ERR("Cannot get IRQ GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->irq_gpio_dev,
+				 CONFIG_PAW3212_IRQ_GPIO_PIN,
+				 GPIO_DIR_IN | GPIO_INT | GPIO_PUD_PULL_UP |
+				 GPIO_INT_LEVEL | GPIO_INT_ACTIVE_LOW);
+	if (err) {
+		LOG_ERR("Cannot configure IRQ GPIO");
+		return err;
+	}
+
+	gpio_init_callback(&dev_data->irq_gpio_cb, irq_handler,
+			   BIT(CONFIG_PAW3212_IRQ_GPIO_PIN));
+
+	err = gpio_add_callback(dev_data->irq_gpio_dev, &dev_data->irq_gpio_cb);
+	if (err) {
+		LOG_ERR("Cannot add IRQ GPIO callback");
+	}
+
+	return err;
+}
+
+static int paw3212_init_spi(struct paw3212_data *dev_data)
+{
+	dev_data->spi_dev = device_get_binding(CONFIG_PAW3212_SPI_DEV_NAME);
+	if (!dev_data->spi_dev) {
+		LOG_ERR("Cannot get SPI device");
+		return -ENXIO;
+	}
+
+	return 0;
+}
+
+static int paw3212_init(struct device *dev)
+{
+	struct paw3212_data *dev_data = &paw3212_data;
+	int err;
+
+	ARG_UNUSED(dev);
+
+	err = paw3212_init_cs(dev_data);
+	if (err) {
+		return err;
+	}
+
+	err = paw3212_init_irq(dev_data);
+	if (err) {
+		return err;
+	}
+
+	err = paw3212_init_spi(dev_data);
+	if (err) {
+		return err;
+	}
+
+	k_delayed_work_init(&dev_data->init_work, paw3212_async_init);
+
+	k_delayed_work_submit(&dev_data->init_work,
+			      async_init_delay[dev_data->async_init_step]);
+
+	return err;
+}
+
+static int paw3212_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct paw3212_data *dev_data = &paw3212_data;
+
+	ARG_UNUSED(dev);
+
+	if (unlikely(!dev_data->ready)) {
+		LOG_INF("Device is not initialized yet");
+		return -EBUSY;
+	}
+
+	int err;
+
+	u8_t motion;
+	err = reg_read(dev_data, PAW3212_REG_MOTION, &motion);
+	if (err) {
+		LOG_ERR("Cannot read X delta");
+		return err;
+	}
+
+	u8_t x_sensor;
+	err = reg_read(dev_data, PAW3212_REG_DELTA_X_L, &x_sensor);
+	if (err) {
+		LOG_ERR("Cannot read X delta");
+		return err;
+	}
+
+	u8_t y_sensor;
+	err = reg_read(dev_data, PAW3212_REG_DELTA_Y_L, &y_sensor);
+	if (err) {
+		LOG_ERR("Cannot read Y delta");
+		return err;
+	}
+
+	if (!err) {
+		s32_t x = (s8_t)x_sensor;
+		s32_t y = (s8_t)y_sensor;
+		dev_data->x = -x;
```

Comment:
Think about Kconfig option informing that sensor is rotated (like for PMW3360)

Reply:
I planned to add it along with attrs but can do it now.